Question 1.Write a program to find the minimum price of ktm bike, 
which are available in Jaipur, and how many ktm bikes are 
available at Jaipur location.

In [2]:
import pandas as pd

In [2]:
df=pd.read_csv("Used_Bikes.csv")
df

bike_name     price       city  kms_driven  \
0        TVS Star City Plus Dual Tone 110cc   35000.0  Ahmedabad     17654.0   
1               Royal Enfield Classic 350cc  119900.0      Delhi     11000.0   
2                      Triumph Daytona 675R  600000.0      Delhi       110.0   
3                      TVS Apache RTR 180cc   65000.0  Bangalore     16329.0   
4      Yamaha FZ S V 2.0 150cc-Ltd. Edition   80000.0  Bangalore     10000.0   
...                                     ...       ...        ...         ...   
32643                Hero Passion Pro 100cc   39000.0      Delhi     22000.0   
32644                  TVS Apache RTR 180cc   30000.0     Karnal      6639.0   
32645              Bajaj Avenger Street 220   60000.0      Delhi     20373.0   
32646             Hero Super Splendor 125cc   15600.0     Jaipur     84186.0   
32647                    Bajaj Pulsar 150cc   22000.0       Pune     60857.0   

             owner   age  power          brand  
0      First Owner   3.0  110.0            TVS  
1      First Owner   4.0  350.0  Royal Enfield  
2      First Owner   8.0  675.0        Triumph  
3      First Owner   4.0  180.0            TVS  
4      First Owner   3.0  150.0         Yamaha  
...            ...   ...    ...            ...  
32643  First Owner   4.0  100.0           Hero  
32644  First Owner   9.0  180.0            TVS  
32645  First Owner   6.0  220.0          Bajaj  
32646  First Owner  16.0  125.0           Hero  
32647  First Owner  13.0  150.0          Bajaj  

[32648 rows x 8 columns]

In [6]:
df1=df[(df["brand"]=="KTM")&(df["city"]=="Jaipur")]
df1

bike_name     price    city  kms_driven        owner  age  power  \
888     KTM RC 390cc  147000.0  Jaipur     15000.0  First Owner  6.0  390.0   
4945  KTM Duke 390cc  200000.0  Jaipur     11700.0  First Owner  4.0  390.0   
4948    KTM RC 390cc  175000.0  Jaipur     10880.0  First Owner  4.0  390.0   
6073    KTM RC 200cc  190000.0  Jaipur      7902.0  First Owner  2.0  200.0   
6159  KTM Duke 250cc  135000.0  Jaipur     12507.0  First Owner  4.0  250.0   
6552    KTM RC 200cc  190000.0  Jaipur      7902.0  First Owner  2.0  200.0   
6734    KTM RC 200cc  128000.0  Jaipur     15000.0  First Owner  4.0  200.0   
7561  KTM Duke 250cc  150000.0  Jaipur     12500.0  First Owner  4.0  250.0   

     brand  
888    KTM  
4945   KTM  
4948   KTM  
6073   KTM  
6159   KTM  
6552   KTM  
6734   KTM  
7561   KTM

In [7]:
df1.shape

(8, 8)

8 ktm bikes are available at jaipur

In [9]:
df1['price'].min()

128000.0

128000.0 is the minimum price of ktm bike, 
which are available in Jaipur

Question 2.Find out how many bikes are available at Delhi location, and 
that are “first owner” bikes. And also after filter the data 
find out the minimum and maximum age of the bike.


In [10]:
df2=df[(df["city"]=="Delhi")&(df["owner"]=="First Owner")]
df2

bike_name     price   city  kms_driven  \
1              Royal Enfield Classic 350cc  119900.0  Delhi     11000.0   
2                     Triumph Daytona 675R  600000.0  Delhi       110.0   
5                         Yamaha FZs 150cc   53499.0  Delhi     25000.0   
6            Honda CB Hornet 160R  ABS DLX   85000.0  Delhi      8200.0   
7      Hero Splendor Plus Self Alloy 100cc   45000.0  Delhi     12645.0   
...                                    ...       ...    ...         ...   
32637                   Bajaj Pulsar 150cc   25000.0  Delhi     32588.0   
32639          Royal Enfield Classic 350cc   95500.0  Delhi     18000.0   
32640               Hero Passion Pro 100cc   32000.0  Delhi     12000.0   
32643               Hero Passion Pro 100cc   39000.0  Delhi     22000.0   
32645             Bajaj Avenger Street 220   60000.0  Delhi     20373.0   

             owner  age  power          brand  
1      First Owner  4.0  350.0  Royal Enfield  
2      First Owner  8.0  675.0        Triumph  
5      First Owner  6.0  150.0         Yamaha  
6      First Owner  3.0  160.0          Honda  
7      First Owner  3.0  100.0           Hero  
...            ...  ...    ...            ...  
32637  First Owner  9.0  150.0          Bajaj  
32639  First Owner  8.0  350.0  Royal Enfield  
32640  First Owner  6.0  100.0           Hero  
32643  First Owner  4.0  100.0           Hero  
32645  First Owner  6.0  220.0          Bajaj  

[6516 rows x 8 columns]

In [11]:
df2.shape

(6516, 8)

6516 are the bikes are available at Delhi location, and 
that are “first owner” bikes

In [12]:
df2['price'].min()

5000.0

In [13]:
df2['price'].max()

1800000.0

Question 3.Try to read excel data using pandas. You can take any excel 
file for this assignment. If you will face any error in this 
question, you can take reference from internet.

In [5]:
data = pd.read_excel(r'C:\Users\Manav Mudgal\Desktop\pyyhon\IV Sem UT-2- Debar List- Display.xlsx')

In [6]:
data

S. No. RTU ROLL NO.  \
0                                                   1   22EJCCS001   
1                                                   2   22EJCCS002   
2                                                   3   22EJCCS003   
3                                                   4   22EJCCS004   
4                                                   5   22EJCCS005   
..                                                ...          ...   
80                                                 65   22EJCCS065   
81                       Number of candidates present          NaN   
82                 Number of candidates absent/ debar          NaN   
83                  Name and Signature of Invigilator          NaN   
84  Total candidates present in IV Semester Sectio...          NaN   

   NAME OF CANDIDATE Room No. Discrete Mathematics Structure  \
0    AADITYA AGRAWAL     LT-1                            NaN   
1      AAYUSH MATHUR      NaN                            NaN   
2        AAYUSH SONI      NaN                            NaN   
3     AAYUSHI SHARMA      NaN                            NaN   
4   ABHIJEET DADHICH      NaN                            NaN   
..               ...      ...                            ...   
80       CHARU RAWAT      NaN                             DB   
81               NaN      NaN                            NaN   
82               NaN      NaN                            NaN   
83               NaN      NaN                            NaN   
84               NaN      NaN                            NaN   

   Technical Communication Microprocessor & Interfaces  \
0                      NaN                         NaN   
1                       db                         NaN   
2                      NaN                         NaN   
3                      NaN                         NaN   
4                       db                         NaN   
..                     ...                         ...   
80                      db                          db   
81                     NaN                         NaN   
82                     NaN                         NaN   
83                     NaN                         NaN   
84                     NaN                         NaN   

   Database Management System Theory of Computation  \
0                         NaN                   NaN   
1                         NaN                    db   
2                         NaN                   NaN   
3                         NaN                   NaN   
4                         NaN                    db   
..                        ...                   ...   
80                         DB                    db   
81                        NaN                   NaN   
82                        NaN                   NaN   
83                        NaN                   NaN   
84                        NaN                   NaN   

   Data Communication and Computer Networks  
0                                       NaN  
1                                       NaN  
2                                       NaN  
3                                       NaN  
4                                       NaN  
..                                      ...  
80                                       DB  
81                                      NaN  
82                                      NaN  
83                                      NaN  
84                                      NaN  

[85 rows x 10 columns]

EXPLORE DATA

In [7]:
data.head(10)

S. No. RTU ROLL NO. NAME OF CANDIDATE Room No.  \
0      1   22EJCCS001   AADITYA AGRAWAL     LT-1   
1      2   22EJCCS002     AAYUSH MATHUR      NaN   
2      3   22EJCCS003       AAYUSH SONI      NaN   
3      4   22EJCCS004    AAYUSHI SHARMA      NaN   
4      5   22EJCCS005  ABHIJEET DADHICH      NaN   
5      6   22EJCCS006    ABHINN AGRAWAL      NaN   
6      7   22EJCCS007    ABHISAR SHARMA      NaN   
7      8   22EJCCS008          ABHISHEK      NaN   
8      9   22EJCCS009     ABHISHEK MODI      NaN   
9     10   22EJCCS010      ADAMYA GUPTA      NaN   

  Discrete Mathematics Structure Technical Communication  \
0                            NaN                     NaN   
1                            NaN                      db   
2                            NaN                     NaN   
3                            NaN                     NaN   
4                            NaN                      db   
5                            NaN                     NaN   
6                            NaN                     NaN   
7                             DB                      db   
8                            NaN                      db   
9                            NaN                     NaN   

  Microprocessor & Interfaces Database Management System  \
0                         NaN                        NaN   
1                         NaN                        NaN   
2                         NaN                        NaN   
3                         NaN                        NaN   
4                         NaN                        NaN   
5                         NaN                        NaN   
6                         NaN                        NaN   
7                          db                         DB   
8                         NaN                         DB   
9                         NaN                        NaN   

  Theory of Computation Data Communication and Computer Networks  
0                   NaN                                      NaN  
1                    db                                      NaN  
2                   NaN                                      NaN  
3                   NaN                                      NaN  
4                    db                                      NaN  
5                   NaN                                      NaN  
6                    db                                      NaN  
7                    db                                       db  
8                   NaN                                       db  
9                   NaN                                      NaN

In [8]:
data.describe()

S. No.  RTU ROLL NO.  NAME OF CANDIDATE  \
count                             85            69                 69   
unique                            70            66                 64   
top     Number of candidates present  RTU ROLL NO.  NAME OF CANDIDATE   
freq                               5             4                  4   

        Room No. Discrete Mathematics Structure Technical Communication  \
count          9                             13                      41   
unique         4                              3                       2   
top     Room No.                             DB                      db   
freq           4                              8                      37   

       Microprocessor & Interfaces Database Management System  \
count                           25                         16   
unique                           3                          3   
top                             db                         DB   
freq                            18                          8   

       Theory of Computation Data Communication and Computer Networks  
count                     36                                       21  
unique                     2                                        3  
top                       db                                       db  
freq                      32                                       13

In [10]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 85 entries, 0 to 84
Data columns (total 10 columns):
 #   Column                                    Non-Null Count  Dtype 
---  ------                                    --------------  ----- 
 0   S. No.                                    85 non-null     object
 1   RTU ROLL NO.                              69 non-null     object
 2   NAME OF CANDIDATE                         69 non-null     object
 3   Room No.                                  9 non-null      object
 4   Discrete Mathematics Structure            13 non-null     object
 5   Technical Communication                   41 non-null     object
 6   Microprocessor & Interfaces               25 non-null     object
 7   Database Management System                16 non-null     object
 8   Theory of Computation                     36 non-null     object
 9   Data Communication and Computer Networks  21 non-null     object
dtypes: object(10)
memory usage: 6.8+ KB


Transform data

In [14]:
data1=data.drop(['Database Management System'],axis=1)
data1

S. No. RTU ROLL NO.  \
0                                                   1   22EJCCS001   
1                                                   2   22EJCCS002   
2                                                   3   22EJCCS003   
3                                                   4   22EJCCS004   
4                                                   5   22EJCCS005   
..                                                ...          ...   
80                                                 65   22EJCCS065   
81                       Number of candidates present          NaN   
82                 Number of candidates absent/ debar          NaN   
83                  Name and Signature of Invigilator          NaN   
84  Total candidates present in IV Semester Sectio...          NaN   

   NAME OF CANDIDATE Room No. Discrete Mathematics Structure  \
0    AADITYA AGRAWAL     LT-1                            NaN   
1      AAYUSH MATHUR      NaN                            NaN   
2        AAYUSH SONI      NaN                            NaN   
3     AAYUSHI SHARMA      NaN                            NaN   
4   ABHIJEET DADHICH      NaN                            NaN   
..               ...      ...                            ...   
80       CHARU RAWAT      NaN                             DB   
81               NaN      NaN                            NaN   
82               NaN      NaN                            NaN   
83               NaN      NaN                            NaN   
84               NaN      NaN                            NaN   

   Technical Communication Microprocessor & Interfaces Theory of Computation  \
0                      NaN                         NaN                   NaN   
1                       db                         NaN                    db   
2                      NaN                         NaN                   NaN   
3                      NaN                         NaN                   NaN   
4                       db                         NaN                    db   
..                     ...                         ...                   ...   
80                      db                          db                    db   
81                     NaN                         NaN                   NaN   
82                     NaN                         NaN                   NaN   
83                     NaN                         NaN                   NaN   
84                     NaN                         NaN                   NaN   

   Data Communication and Computer Networks  
0                                       NaN  
1                                       NaN  
2                                       NaN  
3                                       NaN  
4                                       NaN  
..                                      ...  
80                                       DB  
81                                      NaN  
82                                      NaN  
83                                      NaN  
84                                      NaN  

[85 rows x 9 columns]